src/main/java/uk/gov/ida/verifyserviceprovider/services/MatchingAssertionService.java

```diff
@@ -25,35 +25,34 @@
 
 public class MatchingAssertionService implements AssertionService {
 
-    private final SamlAssertionsSignatureValidator assertionsSignatureValidator;
-    private final AssertionValidator assertionValidator;
 
-    public MatchingAssertionService(
-        SamlAssertionsSignatureValidator assertionsSignatureValidator,
-        AssertionValidator assertionValidator
-    ) {
-        this.assertionsSignatureValidator = assertionsSignatureValidator;
+    private AssertionValidator assertionValidator;
+    private SamlAssertionsSignatureValidator assertionsSignatureValidator;
+
+    public MatchingAssertionService( AssertionValidator assertionValidator, SamlAssertionsSignatureValidator assertionsSignatureValidator ) {
         this.assertionValidator = assertionValidator;
+        this.assertionsSignatureValidator = assertionsSignatureValidator;
     }
 
+
     public TranslatedResponseBody translateSuccessResponse(
-        List<Assertion> assertions,
-        String expectedInResponseTo,
-        LevelOfAssurance expectedLevelOfAssurance,
-        String entityId
+            List<Assertion> assertions,
+            String expectedInResponseTo,
+            LevelOfAssurance expectedLevelOfAssurance,
+            String entityId
     ) {
-        validateAssertions(assertions);
+        //  1. check saml has assertions
+        checkSamlhasAssertions(assertions);
+        //  2. validate assertions
         Assertion assertion = assertions.get(0);
```

Comment:
Yeah that's me trying to understand what was going; wanted to see if this was potentially common functionally between Matching and Non-Matching. In future this would either be removed altogether or moved into there own methods as you suggest.